firmware/defmt-itm/src/lib.rs

```diff
@@ -64,6 +68,14 @@ unsafe impl defmt::Logger for Logger {
         unsafe { ENCODER.start_frame(do_write) }
     }
 
+    unsafe fn flush() {
+        // wait for the queue to be able to accept more data
+        while !stim_0().is_fifo_ready() {}
+
+        // delay "a bit" to drain the queue
+        asm::delay(100); // TODO: how long does it take to drain?
```

Comment:
How long does it take to drain?

Reply:
I think the TPIU can be clocked from an essentially arbitrary source, so there is no reliable way of determining this. Since the ITM does not seem to have any other facilities for flushing its output, it looks like we can't implement `flush` reliably for this transport.

I suppose that means that `flush` is, depending on the defmt transport, either a guaranteed operation, or just advisory.

Reply:
Ah, I see that we've already documented `flush` as "best effort"

Reply:
If we can't determine the timing for itm, I would start with a relatively low one and add a note that people should open bug reports if it is too short in their case. If that happens we can see about raising it.

Reply:
What do you think about the comment I added?

Reply:
Seems good